rules/sensio/src/TypeDeclaration/ReturnTypeDeclarationUpdater.php

```diff
@@ -86,9 +86,13 @@ private function updatePhp(ClassMethod $classMethod, string $className): void
         // change return type
         if ($classMethod->returnType !== null) {
             $returnTypeName = $this->nodeNameResolver->getName($classMethod->returnType);
-            if ($returnTypeName !== null && is_a($returnTypeName, $className, true)) {
+            if ($returnTypeName === null) {
                 return;
             }
+            if (!is_a($returnTypeName, $className, true)) {
+                return;
+            }
+            return;
         }
 
         $classMethod->returnType = new FullyQualified($className);
```

Comment:
@TomasVotruba this looks like an edge case that has two options to be solved:
1. Skip `if` if it contains `return;` (void) statement
2. Process if and replace `if`'s statements with `continue;` statement

Let me know what is the preferred solution in your opinion.

Reply:
I think 1. is right in this case.
What do you think?

Reply:
For this particular rule option #1 makes more sense to me as well.